src/altinn-app-frontend/src/components/presentation/NavBar.test.tsx

```diff
@@ -1,26 +1,74 @@
-import { mount } from 'enzyme';
 import React from 'react';
-import NavBar from './NavBar';
+import NavBar, { INavBarProps } from './NavBar';
+import { screen } from '@testing-library/react';
+import { getLanguageFromCode } from 'altinn-shared/language';
+import userEvent from '@testing-library/user-event';
+import { IAppLanguage } from 'altinn-shared/types';
+import { renderWithProviders } from '../../../testUtils';
+
+const renderNavBar = (props?: Partial<INavBarProps>) => {
+  const mockClose = jest.fn();
+  const mockBack = jest.fn();
+  const mockAppLanguageChange = jest.fn();
+  const mockAppLanguages: IAppLanguage[] = [
+    {
+      language: 'nb',
+    },
+    {
+      language: 'en',
+    },
+  ];
+  renderWithProviders(
+    <NavBar
+      language={getLanguageFromCode('nb')}
+      handleClose={mockClose}
+      handleBack={mockBack}
+      appLanguages={mockAppLanguages}
+      onAppLanguageChange={mockAppLanguageChange}
+      selectedAppLanguage={'nb'}
+      {...props}
+    />,
+  );
+
+  return { mockClose, mockBack, mockAppLanguageChange };
+};
 
 describe('components/presentation/NavBar.tsx', () => {
-  it('should render back button by default', () => {
-    const wrapper = mount(
-      <NavBar language={{}} handleClose={null} handleBack={null} />,
-    );
-    expect(wrapper.find('div.a-modal-navbar')).toHaveLength(1);
-    expect(wrapper.find('button.a-modal-close')).toHaveLength(1);
+  it('should render close button', async () => {
+    const { mockClose } = renderNavBar();
+    const closeButton = screen.getByRole('button', { name: /Lukk Skjema/i });
+    // Should not render back button
```

Comment:
I'm probably misunderstanding, or missing some context, but I dont understand this comment. It mentions a "back button", but the test title is mentioning a "close button". Are they the same thing? Theres nothing in the test that verifies that a "back button" is not rendered?

Reply:
Well, the comment points out that since line 39 checks that the close button is rendered and line 41 checks that there is in total 1 button, this means that the back button is not rendered (or that the close button is the only "button" rendered). Maybe not relevent to point out.

Reply:
I see. Could I suggest to instead add an extra assertion for this instead of the comment?

Something like this:

```suggestion
    expect(screen.queryByTestId('altinn-back-button')).not.toBeInTheDocument();
```

This way its explicitly part of the test, and not a comment that someone may or may not read, and have to interpret (and misunderstand like I did) 🙂